src/mmds/src/data_store.rs

```diff
@@ -190,16 +190,21 @@ impl Mmds {
         Ok(())
     }
 
-    // We do not check size of data_store before returning a result because due
-    // to limit from put/patch the data_store can not be bigger than the limit
-    // imposed by the server.
-    pub fn get_data_str(&self) -> String {
+    #[cfg(test)]
+    fn get_data_str(&self) -> String {
         if self.data_store.is_null() {
             return String::from("{}");
         }
         self.data_store.to_string()
     }
 
+    // We do not check size of data_store before returning a result because due
+    // to limit from put/patch the data_store can not be bigger than the limit
+    // imposed by the server.
+    pub fn data_store(&self) -> Value {
```

Comment:
nit: I would rename this to `data_store_value` or something like this. `data_store` implies you get a reference to the data store.